include/zephyr/net/ethernet.h

```diff
@@ -433,8 +436,11 @@ enum ethernet_if_types {
 	/** IEEE 802.3 Ethernet (default) */
 	L2_ETH_IF_TYPE_ETHERNET,
 
-	/** IEEE 802.11 Wi-Fi*/
-	L2_ETH_IF_TYPE_WIFI,
+	/** IEEE 802.11 Wi-Fi Station */
+	L2_ETH_IF_TYPE_WIFI_STA,
+
+	/** IEEE 802.11 Wi-Fi Soft AP */
+	L2_ETH_IF_TYPE_WIFI_UAP,
```

Comment:
This `ethernet_if_type` is used to differentiate Wi-Fi from Ethernet. The Station vs AP selection is a wifi property and this check does not really belong here.

Reply:
Also, FYI, I have been SAP to represent SoftAP

Reply:
Hi @jukkar, split L2_ETH_IF_TYPE_WIFI into L2_ETH_IF_TYPE_WIFI_STA and L2_ETH_IF_TYPE_WIFI_SAP is a more easy and clean way. Or can I add new "enum wifi_if_types" in wifi.h and add "enum wifi_if_types wifi_if_type" in "struct ethernet_context"? do you have any suggesions?

Reply:
Hi @krish2718 , I will change UAP to SAP, thanks

Reply:
Hi @krish2718, UAP is the abbreviation of Micro-AP, can we keep this unchanged?

Reply:
Hi @jukkar @krish2718 
about the UAP, I can change it to SAP.
can I add new "enum wifi_if_types"(contains STA and SAP)  in wifi.h? and add "enum wifi_if_types wifi_if_type" in "struct ethernet_context", and keep "enum ethernet_if_types eth_if_type" unchanged?
do you have any suggestions to implement the "STA and SAP"？

Reply:
So, far the Wi-Fi management doesn't have any state, it's mostly a relay from applications to the drivers, but now we have a case where the NM (i.e., WPA supplicant) needs to store per-iface data, so, IMHO, instead of putting this in L2-Ethernet context, we can introduce NM private data?

Reply:
Hi @krish2718 , I try to introduce NM private data, you mean in this?
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151499687/2bce3acf-0eae-4cf4-b68a-0496b97d916f)
but I have a problem, in add_interface(), it will call wifi_nm_register_mgd_iface(), if I also add the sta/softap type here, based on the net_if iface, how could I know this interface belongs to sta or softap?

Reply:
Hi @krish2718 can I just add enum wifi_if_types"(contains STA and SAP) into struct ethernet_context, and keep enum ethernet_if_types unchanged? it's more easy and clean to implement it.

Reply:
> have a case where the NM (i.e., WPA supplicant) needs to store per-iface data

I was proposing per-face data, basically a wrapper around `struct net_if`, something like below (just a proposal).

```
enum iface_type {
          STA = 0,
          SAP,
          P2P_CLI,
          P2P_GO
};
struct nm_mgd_iface {
   unsigned char type; /* bitmap of types, 8 types should suffice or we can make it short */
   struct net_if *iface;
};

struct nm_mgd_iface *mgd_ifaces[];

char nm_wifi_get_type(void *priv);
```

WDYT? Does this solve your problem?

Reply:
Hi @krish2718 
can you review all again, thanks